### PR TITLE
Feat 2.1.x/db#65456 sorting fields widget option improvements

### PR DESCRIPTION
**Pull request by NathanHGit**
# Description

I removed the sortable boolean from the application settings object and I made sure that the sort resets when changing display mode. 

I also fixed some bugs:
- an error was created when event was null on onSort function
- the sort was not taken into account when searching for something
- the pagination was incorrect when you were in a different page than the first and searching for something (pagination was not reset)

## Ticket

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/65456

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- I created a summary cards widget
- I selected the "Allow sorting" button in the widget display panel from widget options
- I added at least one field corresponding to the current layout in the list of the sorting panel
- I sorted my widget
- I changed display mode to "grid", and came back to "cards"
- I checked if sort select had been reset and the cards were in the correct order

## Sreenshots

peek

# Checklist:

( * == Mandatory ) 

- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by unai-reliefapp** [comment]
Hi @NathanHGit Please format the code correctly and push it again in order to see real changes and be able to review it correctly
Thanks

**Review by RenzoPrats — APPROVED** [review]
Working fine as expected, looks good to me!

File: libs/safe/src/lib/components/widgets/summary-card/summary-card.component.html [hunk 1]
```diff
@@ -130,9 +131,10 @@
 
 <!-- Selection of sort -->
 <ng-template #sortTemplate>
-  <ng-container *ngIf="sortFields.length > 1">
+  <ng-container *ngIf="sortFields.length > 0">
     <div uiFormFieldDirective class="!m-0 w-44">
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/widgets/summary-card/summary-card.component.html]:
```suggestion
    <div uiFormFieldDirective class="items-center flex w-44">
```

From this:
image

To this:
image

**Review by unai-reliefapp — CHANGES_REQUESTED** [review]

**Comment by unai-reliefapp** [comment]
Looks good to me, also the use of the skip property from the event instead of calculated from the paginator component is a fine improvement 👍

**Pull request merged**